File: app/src/main/java/com/example/nutritionapp/Register/SharedViewModel.java
package com.example.nutritionapp.Register;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class SharedViewModel extends ViewModel {
    private MutableLiveData<String> email = new MutableLiveData<>();
    private MutableLiveData<String> pass = new MutableLiveData<>();
    private MutableLiveData<String> name = new MutableLiveData<>();
    private MutableLiveData<String> mobile = new MutableLiveData<>();
    private MutableLiveData<String> age = new MutableLiveData<>();
    private MutableLiveData<String> height = new MutableLiveData<>();
    private MutableLiveData<String> weight = new MutableLiveData<>();
    private MutableLiveData<String> gender = new MutableLiveData<>();
    private MutableLiveData<String> active = new MutableLiveData<>();
    private MutableLiveData<Double> calories = new MutableLiveData<>();
    private double a;
    private double calorie;

    public void setCalories(String height, String weight, String age, String gender, String active){
        if(active.equals("Light"))
            a= 1.2;
        else if(active.equals("Moderate"))
            a= 1.55;
        else
            a= 1.9;
        if(gender.equals("Male"))
            calorie= (66.5 + 13.8 * Integer.parseInt(weight) + 5* Integer.parseInt(height) ) - (6.8 * Integer.parseInt(age)) * a;
        else
            calorie= (655.1 + 9.5 * Integer.parseInt(weight) + 1.8* Integer.parseInt(height) ) - (4.6* Integer.parseInt(age)) * a;
        calories.setValue(calorie);

    }

    public void setEmailMobile(String input1, String input2) {
        email.setValue(input1);
        mobile.setValue(input2);
    }

    public void setPass(String input3){
        pass.setValue(input3);
    }

    public void setName(String input4){
        name.setValue(input4);
    }

    public void setAge(String input5){
        age.setValue(input5);
    }

    public void setHeight(String input6){
        height.setValue(input6);
    }

    public void setWeight(String input7){
        weight.setValue(input7);
    }

    public void setGender(String input8){
        gender.setValue(input8);
    }

    public void setActive(String input9){
        active.setValue(input9);
    }

    public MutableLiveData<String> getPass() {
        return pass;
    }

    public MutableLiveData<String> getName() {
        return name;
    }

    public MutableLiveData<String> getAge() {
        return age;
    }

    public MutableLiveData<String> getHeight() {
        return height;
    }

    public MutableLiveData<String> getWeight() {
        return weight;
    }

    public MutableLiveData<String> getGender() {
        return gender;
    }

    public MutableLiveData<String> getActive() {
        return active;
    }

    public LiveData<String> getEmail() {
        return email;
    }

    public LiveData<String> getMobile() {
        return mobile;
    }

    public MutableLiveData<Double> getCalories() {
        return calories;
    }
}

File: app/src/main/java/com/example/nutritionapp/Register/SharedViewModel.kt
package com.example.nutritionapp.Register

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class SharedViewModel : ViewModel() {
    private val email = MutableLiveData<String>()
    val pass = MutableLiveData<String>()
    val name = MutableLiveData<String>()
    private val mobile = MutableLiveData<String>()
    val age = MutableLiveData<String>()
    val height = MutableLiveData<String>()
    val weight = MutableLiveData<String>()
    val gender = MutableLiveData<String>()
    val active = MutableLiveData<String>()
    val calories = MutableLiveData<Double>()
    private var a = 0.0
    private var calorie = 0.0
    fun setCalories(height: String, weight: String, age: String, gender: String, active: String) {
        a = if (active == "Light") 1.2 else if (active == "Moderate") 1.55 else 1.9
        calorie =
            if (gender == "Male") 66.5 + 13.8 * weight.toInt() + 5 * height.toInt() - 6.8 * age.toInt() * a else 655.1 + 9.5 * weight.toInt() + 1.8 * height.toInt() - 4.6 * age.toInt() * a
        calories.value = calorie
    }

    fun setEmailMobile(input1: String, input2: String) {
        email.value = input1
        mobile.value = input2
    }

    fun setPass(input3: String) {
        pass.value = input3
    }

    fun setName(input4: String) {
        name.value = input4
    }

    fun setAge(input5: String) {
        age.value = input5
    }

    fun setHeight(input6: String) {
        height.value = input6
    }

    fun setWeight(input7: String) {
        weight.value = input7
    }

    fun setGender(input8: String) {
        gender.value = input8
    }

    fun setActive(input9: String) {
        active.value = input9
    }

    fun getEmail(): LiveData<String> {
        return email
    }

    fun getMobile(): LiveData<String> {
        return mobile
    }
}